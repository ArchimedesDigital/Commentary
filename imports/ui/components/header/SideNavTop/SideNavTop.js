import React from 'react';

// components:
import AvatarIcon from '/imports/ui/components/avatar/AvatarIcon'; // eslint-disable-line import/no-absolute-path

const SideNavTop = ({ currentUser, username }) => (
	<div className="sidenav-top">
		{Meteor.user() &&
			<a href="/profile">
				<div className="user-image paper-shadow">
					<AvatarIcon avatar={currentUser && currentUser.profile ? currentUser.profile.avatarUrl : '/images/default_user.jpg'} />
				</div>
			</a>}
		<a href="/profile">
			<span className="user-fullname">
				{username}
			</span>
		</a>
	</div>
);
SideNavTop.propTypes = {
	currentUser: React.PropTypes.shape({
		profile: React.PropTypes.shape({
			avatarUrl: React.PropTypes.string,
		}),
	}),
	username: React.PropTypes.string,
};
SideNavTop.defaultProps = {
	currentUser: null,
	username: null,
};

export default SideNavTop;
