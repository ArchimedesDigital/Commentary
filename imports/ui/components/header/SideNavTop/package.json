{
  "name": "SideNavTop",
  "version": "0.0.1",
  "private": true,
  "main": "./SideNavTop.js"
}
