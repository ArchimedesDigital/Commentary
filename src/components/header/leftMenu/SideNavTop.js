import React from 'react';
import PropTypes from 'prop-types';
import Cookies from 'js-cookie';
import { Link } from 'react-router-dom';

// components:
import AvatarIcon from '../../../modules/profile/avatar/AvatarIcon';

const SideNavTop = ({ currentUser, username }) => (
	<div className="sidenav-top">
		{Cookies.getItem('user') &&
			<Link to="/profile">
				<div className="user-image paper-shadow">
					<AvatarIcon avatar={currentUser && currentUser.profile ? currentUser.profile.avatarUrl : '/images/default_user.jpg'} />
				</div>
			</Link>}
		<Link to="/profile">
			<span className="user-fullname">
				{username}
			</span>
		</Link>
	</div>
);
SideNavTop.propTypes = {
	currentUser: PropTypes.shape({
		profile: PropTypes.shape({
			avatarUrl: PropTypes.string,
		}),
	}),
	username: PropTypes.string,
};
SideNavTop.defaultProps = {
	currentUser: null,
	username: null,
};

export default SideNavTop;
