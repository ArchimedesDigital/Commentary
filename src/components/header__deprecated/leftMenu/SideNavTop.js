import React from 'react';
import PropTypes from 'prop-types';
import Cookies from 'js-cookie';
import { Link } from 'react-router-dom';

// components:
import AvatarIcon from '../../../modules/profile/avatar/AvatarIcon';

const SideNavTop = ({ currentUser, username }) => (
	<div className="sidenav-top">
		{Cookies.get('user') &&
			<Link to="/profile">
				<div className="user-image paper-shadow">
					<AvatarIcon avatar={currentUser && currentUser.profile ? currentUser.profile.avatarUrl : '/images/default_user.jpg'} />
				</div>
			</Link>}
		<Link to="/profile">
			<span className="user-fullname">
				{username}
			</span>
		</Link>
	</div>
);
SideNavTop.propTypes = {
	currentUser: PropTypes.string,
	username: PropTypes.string,
};
SideNavTop.defaultProps = {
	currentUser: null,
	username: null,
};

export default SideNavTop;
