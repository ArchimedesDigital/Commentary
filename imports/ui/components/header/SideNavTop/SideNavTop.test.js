import React from 'react';
import renderer from 'react-test-renderer';

// component:
import SideNavTop from './SideNavTop';

describe('SideTopNav', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<SideNavTop />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
