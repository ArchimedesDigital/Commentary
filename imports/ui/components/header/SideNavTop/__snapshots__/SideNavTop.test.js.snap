// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideTopNav renders correctly 1`] = `
<div
  className="sidenav-top"
>
  <a
    href="/profile"
  >
    <span
      className="user-fullname"
    />
  </a>
</div>
`;
